packages/interactivity/src/store.ts

```diff
@@ -34,25 +32,34 @@ export const getConfig = ( namespace?: string ) =>
  *
  * @example
  * ```js
- *  const { state } = store('myStore', {
+ *  const { state } = store( 'myPlugin', {
  *    callbacks: {
  *      updateServerState() {
  *        const serverState = getServerState();
  *        // Override some property with the new value that came from the server.
  *        state.overridableProp = serverState.overridableProp;
  *      },
  *    },
- *  });
+ *  } );
  * ```
  *
- * @param namespace Store's namespace from which to retrieve the server state.
+ * @param namespace Store namespace. By default, it inherits the namespace of
+ *                  the store where it is defined.
  * @return The server state for the given namespace.
  */
-export const getServerState = ( namespace?: string ) => {
+export const getServerState: ( < T extends object >(
+	namespace?: string
+) => DeepReadonly< T > ) & { subscribe?: number } = < T extends object >(
+	namespace?: string
+): DeepReadonly< T > => {
```

Comment:
Maybe this can be simplified. Do you know if the assigned value also needs to be typed?


```suggestion
) => DeepReadonly< T > ) & { subscribe?: number } = ( namespace ) => {
```

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/72381/commits/12ad5350c45cf8fa7ab3b2578d67231810e8406d.